ui/lib/components/TopLoadingBar/index.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useRef, useEffect } from 'react'
+import LoadingBar from 'react-top-loading-bar'
+
+const useLoadingBar = () => {
+  const loadingBar = useRef<LoadingBar>(null)
+
+  useEffect(() => {
+    function startLoading() {
+      loadingBar?.current?.continuousStart()
+    }
+    window.addEventListener('single-spa:before-routing-event', startLoading)
```

Comment:
maybe better to use [`useEventListener`](https://hooks.umijs.org/dom/use-event-listener)

Reply:
Let me have a try.

Reply:
Nice! works fine and code is much concise than before.